### PR TITLE
add customizer strings upload (WP-579)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php [hunk 1]
```diff
@@ -158,7 +158,6 @@ protected function getFieldNameByMethodName($method)
      */
     public function __call($method, array $params)
     {
-        $this->getLogger()->debug("Magic call to $method");
```
Comment (vsolovei-smartling) [inline inc/Smartling/DbAl/WordpressContentEntities/EntityAbstract.php:161]: This has served it's purpose

File: inc/Smartling/Base/SmartlingCoreUploadTrait.php [hunk 2]
```diff
@@ -232,7 +233,7 @@ public function applyXML(SubmissionEntity $submission, string $xml, XmlHelper $x
             $targetContent = $this->getContentHelper()->readTargetContent($submission);
             $params = new AfterDeserializeContentEventParameters($translation, $submission, $targetContent, $translation['meta']);
             do_action(ExportedAPI::EVENT_SMARTLING_AFTER_DESERIALIZE_CONTENT, $params);
-            $this->applyBlockLevelLocks($targetContent, $translation, $submission, $postContentHelper, $lockedData['entity']);
+            $targetContent = $this->applyBlockLevelLocks($targetContent, $translation, $submission, $postContentHelper, $lockedData['entity']);
```
Comment (vsolovei-smartling) [inline inc/Smartling/Base/SmartlingCoreUploadTrait.php:236]: Expect $targetContent to be changed, even without reading fn phpdoc

File: inc/Smartling/Base/SmartlingCoreUploadTrait.php [hunk 3]
```diff
@@ -753,22 +754,24 @@ private function removeExcludedFields(array $fields, ConfigurationProfileEntity
...
+    private function applyBlockLevelLocks(EntityAbstract $targetContent, array $translation, SubmissionEntity $submission, PostContentHelper $postContentHelper, array $entity): EntityAbstract
     {
         if (array_key_exists('entity', $translation) && ArrayHelper::notEmpty($translation['entity'])) {
             $targetContentArray = $targetContent->toArray();
             if (array_key_exists('post_content', $translation['entity']) && array_key_exists('post_content', $targetContentArray) && count($submission->getLockedFields()) > 0) {
                 $translation['entity']['post_content'] = $postContentHelper->applyTranslation($targetContentArray['post_content'], $translation['entity']['post_content'], $submission->getLockedFields());
             }
             $translation['entity'] = self::arrayMergeIfKeyNotExists($entity, $translation['entity']);
-            $this->setValues($targetContent, $translation['entity']);
+            $targetContent = $this->setValues($targetContent, $translation['entity']);
```
Comment (vsolovei-smartling) [inline inc/Smartling/Base/SmartlingCoreUploadTrait.php:765]: Again, expect changes to the variable

File: inc/Smartling/Bootstrap.php [hunk 4]
```diff
@@ -424,20 +418,19 @@ protected function testPluginSetup()
...
         ContentTypeNavigationMenu::register($di);
 
-        $handlers = [
-            'taxonomies' => (new Taxonomies($di)),
-            'posts'      => (new PostTypes($di)),
-        ];
+        // register taxonomies
+        new Taxonomies($di);
+        // register post types
+        new PostTypes($di);
```
Comment (vsolovei-smartling) [inline inc/Smartling/Bootstrap.php:433]: A call to new Taxonomies() and new PostTypes is required at bootstrap, but having an array of handlers that is never used is not

Comment (PavelLoparev) [reply inc/Smartling/Bootstrap.php:433]: Nice catch. But it's strange that we need just instantiate object to register it somewhere somehow though.

File: inc/Smartling/ContentTypes/ContentTypeCustomizer.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    public function registerIOWrapper(): void
+    {
+        $di = $this->getContainerBuilder();
+        $wrapperId = 'wrapper.entity' . $this->getSystemName();
+        $definition = $di->register($wrapperId, CustomizerEntity::class);
+        $definition
+            ->addArgument($this->getSystemName())
+            ->addArgument([]);
```
Comment (PavelLoparev) [inline inc/Smartling/ContentTypes/ContentTypeCustomizer.php:52]: Why do we register service manually?

File: inc/Smartling/Helpers/ContentHelper.php [hunk 6]
```diff
@@ -328,6 +322,9 @@ public function writeTargetContent(SubmissionEntity $submission, EntityAbstract
             throw new \LogicException($message, $e->getCode(), $e);
         }
 
+        if ($result instanceof EntityAbstract) {
+            return $result->getPK();
+        }
```
Comment (PavelLoparev) [inline inc/Smartling/Helpers/ContentHelper.php:327]: Didn't get the reason for this change

File: inc/Smartling/WP/View/BulkSubmit.php [hunk 7]
```diff
@@ -101,30 +99,31 @@
...
-            <input type="hidden" name="page" value="<?= $_REQUEST['page']; ?>"/>
+            <input type="hidden" name="wp_customize" value="on"/>
+            <input type="hidden" name="page" value="<?= $_REQUEST['page'] ?>"/>
             <input type="hidden" id="action" name="action" value="clone"/>
     </form>
 
     <script>
         (function ($) {
-            $(document).ready(function () {
+            $(document).on('ready', function () {
```
Comment (PavelLoparev) [inline inc/Smartling/WP/View/BulkSubmit.php:124]: Isn't `.ready` and alias to `.on('ready')`?

**Review by PavelLoparev — APPROVED** [review]

**Comment by dimitrystd** [comment]
Close PR without merge. We will have to review branch from the beginning when will be ready to work on this feature

**Pull request closed without merging**